### PR TITLE
Issue #1327 Improve execution plans for search

### PR DESCRIPTION
**Pull request by punktilious**

### CODE REVIEW

File: fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCQueryBuilder.java [hunk 1]
```diff
@@ -648,8 +648,13 @@ protected SqlQueryData processChainedReferenceParm(QueryParameter queryParm) thr
             } else {
                 // This logic processes the LAST parameter in the chain.
                 // Build this piece: CPx.PARAMETER_NAME_ID = x AND CPx.STR_VALUE = ?
+                if (chainedParmVar == null) {
+                    chainedParmVar            = CP + 1;
+                }
                 Class<?> chainedResourceType = ModelSupport.getResourceType(resourceTypeName);
                 SqlQueryData sqlQueryData = buildQueryParm(chainedResourceType, currentParm, chainedParmVar);
+
+                // log.info("chained sqlQueryData[" + chainedParmVar + "] = " + sqlQueryData.getQueryString());
```
Comment (prb112) [inline fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCQueryBuilder.java:657]: I'm ok with it left there, probably should wrap in is loggable fine and log.fine

File: fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/type/LocationParmBehaviorUtil.java [hunk 2]
```diff
@@ -46,7 +46,7 @@ public LocationParmBehaviorUtil() {
      * @param boundingAreas
      */
     public void buildLocationSearchQuery(String populateNameIdSubSegment, StringBuilder whereClauseSegment,
-            List<Object> bindVariables, List<Bounding> boundingAreas) {
```
Comment (prb112) [inline fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/type/LocationParmBehaviorUtil.java:49]: Copyright 2019

File: fhir-persistence-jdbc/src/test/java/com/ibm/fhir/persistence/jdbc/test/util/LocationParmBehaviorUtilTest.java [hunk 3]
```diff
@@ -37,7 +39,7 @@ private void runTestBoundingBox(List<Object> expectedBindVariables, String expec
         List<Object> actualBindVariables = new ArrayList<>();
 
         LocationParmBehaviorUtil util = new LocationParmBehaviorUtil();
-        util.buildQueryForBoundingBox(actualWhereClauseSegment, actualBindVariables, boundingBox);
+        util.buildQueryForBoundingBox(actualWhereClauseSegment, actualBindVariables, boundingBox, JDBCConstants.PARAMETER_TABLE_ALIAS);
```
Comment (prb112) [inline fhir-persistence-jdbc/src/test/java/com/ibm/fhir/persistence/jdbc/test/util/LocationParmBehaviorUtilTest.java:42]: 2020 copyright

File: fhir-persistence/src/main/java/com/ibm/fhir/persistence/util/AbstractQueryBuilder.java [hunk 4]
```diff
@@ -233,7 +233,7 @@ protected T1 buildQueryParm(Class<?> resourceType, QueryParameter queryParm)
      * @return T1 - A query segment related to a LocationPosition
      * @throws FHIRPersistenceException
      */
-    protected T1 processLocationPosition(List<QueryParameter> queryParameters) throws FHIRPersistenceException {
+    protected T1 processLocationPosition(List<QueryParameter> queryParameters, String paramTableAlias) throws FHIRPersistenceException {
```
Comment (prb112) [inline fhir-persistence/src/main/java/com/ibm/fhir/persistence/util/AbstractQueryBuilder.java:236]: Copyright

File: fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCQueryBuilder.java (hunk 1 above)
Comment (punktilious) [reply fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCQueryBuilder.java]: wrapped and changed to fine

File: fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/type/LocationParmBehaviorUtil.java (hunk 2 above)
Comment (punktilious) [reply fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/type/LocationParmBehaviorUtil.java:49]: fixed

File: fhir-persistence-jdbc/src/test/java/com/ibm/fhir/persistence/jdbc/test/util/LocationParmBehaviorUtilTest.java (hunk 3 above)
Comment (punktilious) [reply fhir-persistence-jdbc/src/test/java/com/ibm/fhir/persistence/jdbc/test/util/LocationParmBehaviorUtilTest.java:42]: fixed

File: fhir-persistence/src/main/java/com/ibm/fhir/persistence/util/AbstractQueryBuilder.java (hunk 4 above)
Comment (punktilious) [reply fhir-persistence/src/main/java/com/ibm/fhir/persistence/util/AbstractQueryBuilder.java:236]: fixed

**Pull request merged**